pygeo/DVConstraints.py

```diff
@@ -221,21 +237,52 @@ def setSurface(self, surf):
         >>> DVCon.setSurface(surf)
 
         """
-
-        if type(surf) == list:
-            # Data from ADflow
-            self.p0 = numpy.array(surf[0])
-            self.v1 = numpy.array(surf[1])
-            self.v2 = numpy.array(surf[2])
-
-        elif isinstance(surf, str):
-            # Load the surf as a plot3d file
-            self.p0, self.v1, self.v2 = self._readPlot3DSurfFile(surf)
-
-        else: # Assume it's a pyGeo surface
-            self._generateDiscreteSurface(surf)
-
-    def setDVGeo(self, DVGeo):
+        if name in self.surfaces.keys():
+            raise KeyError('Surface names must be unique. Repeated surface name: ' + str(name))
+
+        self.surfaces[name] = list()
+        if format == 'point-vector':
+            if type(surf) == list:
+                # Data from ADflow
+                p0 = numpy.array(surf[0])
+                v1 = numpy.array(surf[1])
+                v2 = numpy.array(surf[2])
+            elif isinstance(surf, str):
+                # Load the surf as a plot3d file
+                p0, v1, v2 = self._readPlot3DSurfFile(surf)
+
+            else: # Assume it's a pyGeo surface
```

Comment:
@joanibal the reason it has to be called twice is that the KS function needs to know the actual minimum distance a priori so it can subtract that off and make the "real" KS computation better conditioned. the same code computes both which is why it gets called twice